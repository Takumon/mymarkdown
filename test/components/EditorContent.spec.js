import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex, { mapState, mapActions } from 'vuex'
import vuescroll from 'vue-scroll'
import {
  MdChips,
} from 'vue-material/dist/components'

import TestTargetVue from '../../components/EditorContent.vue'

jest.useFakeTimers()

describe('EditContent.vue', () => {
  let wrapper
  let store
  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(MdChips)
    localVue.use(vuescroll)
    localVue.use(Vuex)

    store = new Vuex.Store({
      state: {
        textEditorPreviewMode: 'tab-editor-and-preview',
      },
      actions: {
        setTextEditorPreviewMode ({ commit }, mode) {
          return new Promise((resolve, reject) => {
            commit( 'setTextEditorPreviewMode', { mode } )
            resolve()
          })
        },
      },
      mutations: {
        setTextEditorPreviewMode (state, payload) {
          state.textEditorPreviewMode = payload.mode
        },
      }
    })

    const sysdate = new Date(2018,1,1,12,0)
    const memo = {
      markdown: '# マークダウン形式のメモ' + `
* 結構
* たくさん
* めもる
* [ ] チェックボックスその１
* [x] チェックボックスその２

\`\`\`
public static void main(String[] args) {
  System.out.println("Test!");
}
\`\`\`


\`\`\`Java
public static void main(String[] args) {
  System.out.println("Test!");
}
\`\`\`
`,
      tags: ['Java', 'JavaScript'],
      created: sysdate,
      updated: sysdate,
    }

    wrapper = shallowMount(TestTargetVue, {
      store,
      localVue,
      propsData: { memo },
    })
    jest.runAllTimers()
  })

  describe('初期表示', () => {
    test('スナップショット', () => {
      expect(wrapper.html().replace(/"md-chips-[^"]+"/, '"md-chips-tempid"')).toMatchSnapshot()
    })
  })

  describe('タブモードをプレビューにした時', () => {
    beforeEach(() => {
      store.dispatch('setTextEditorPreviewMode', 'tab-preview')
      jest.runAllTimers()
    })

    test('編集部分は表示されていなくて、プレビュー部分が表示されている、', () => {
      const editorContent = wrapper.find('[data-test="editor-content"]')
      expect(editorContent.classes()).toContain('tab-preview')
    })
  })

  describe('タブモードをエディターにした時', () => {
    beforeEach(() => {
      store.dispatch('setTextEditorPreviewMode', 'tab-editor')
      jest.runAllTimers()
    })

    test('編集部分は表示されていて、プレビュー部分が表示されていない', () => {
      const editorContent = wrapper.find('[data-test="editor-content"]')
      expect(editorContent.classes()).toContain('tab-editor')
    })
  })

  describe('キーアップ時', () => {
    beforeEach(() => {
      const editor = wrapper.find('[data-test="editor"]')
      editor.trigger('keyup', {
        ctrlKey: true,
        witch: 83,
        keyCode: 83
      })
      jest.runAllTimers()
    })

    test('saveMemoイベントが発行されている', () => {
      expect(wrapper.emitted('saveMemo')).toBeTruthy()
    })
  })





})