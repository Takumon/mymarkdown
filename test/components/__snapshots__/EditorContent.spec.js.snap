// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditContent.vue 初期表示 スナップショット 1`] = `
<div>
  <div class="memo-info-area">
    <div class="md-field md-chips md-theme-default md-has-placeholder md-theme-default md-has-value">
      <div tabindex="0" class="md-chip md-theme-default md-deletable md-clickable md-chip-enter md-chip-enter-active">
        <div class="md-ripple">Java </div>
        <button type="button" class="md-button md-icon-button md-dense md-input-action md-clear md-theme-default md-input-action-enter md-input-action-enter-active" tabindex="-1">
          <div class="md-ripple">
            <div class="md-button-content">
              <i class="md-icon md-icon-font md-icon-image md-theme-default">
                <svg height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
                  <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path>
                  <path d="M0 0h24v24H0z" fill="none"></path>
                </svg>
              </i>
            </div>
          </div>
        </button>
      </div>
      <div tabindex="0" class="md-chip md-theme-default md-deletable md-clickable md-chip-enter md-chip-enter-active">
        <div class="md-ripple">JavaScript </div>
        <button type="button" class="md-button md-icon-button md-dense md-input-action md-clear md-theme-default md-input-action-enter md-input-action-enter-active" tabindex="-1">
          <div class="md-ripple">
            <div class="md-button-content">
              <i class="md-icon md-icon-font md-icon-image md-theme-default">
                <svg height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
                  <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path>
                  <path d="M0 0h24v24H0z" fill="none"></path>
                </svg>
              </i>
            </div>
          </div>
        </button>
      </div>
      <input type="text" id="md-chips-tempid" placeholder="Add Tag..." class="md-input">
      <!---->
      <!---->
      <!---->
    </div>
  </div>
  <div data-test="editor-content" class="editor tab-editor-and-preview">
    <textarea data-test="editor" class="markdown"></textarea>
    <div data-test="preview" id="preview" class="preview markdown-body">
      <h1 id="-">マークダウン形式のメモ</h1>
      <ul>
        <li>結構</li>
        <li>たくさん</li>
        <li>めもる</li>
        <li>
          <input disabled="" type="checkbox"> チェックボックスその１</li>
        <li>
          <input checked="" disabled="" type="checkbox"> チェックボックスその２</li>
      </ul>
      <pre><code>public static void main(String[] args) {
  System.out.println("Test!");
}</code></pre><pre><code class="Java"><span class="hljs-function"><span class="hljs-keyword">public</span> <span class="hljs-keyword">static</span> <span class="hljs-keyword">void</span> <span class="hljs-title">main</span><span class="hljs-params">(String[] args)</span> </span>{
  System.out.println(<span class="hljs-string">"Test!"</span>);
}</code></pre>
    </div>
  </div>
  <div class="characters">Characters 239</div>
  <div class="date-area">
    <div>Updated: "2018-02-01T03:00:00.000Z"</div>
    <div>Created: "2018-02-01T03:00:00.000Z"</div>
  </div>
</div>
`;
